src/integrationTest/java/com/mongodb/hibernate/BasicCrudIntegrationTests.java

```diff
@@ -198,6 +198,27 @@ void testDynamicUpdate() {
         }
     }
 
+    @Nested
+    class LoadByPrimaryKeyTests {
+
+        @Test
+        void testGetById() {
+            var book = new Book();
+            book.id = 1;
+            book.title = "In Search of Lost Time";
+            book.publishYear = 1913;
+
+            sessionFactoryScope.inTransaction(session -> session.persist(book));
+
+            var loadedBook = sessionFactoryScope.fromTransaction(session -> session.get(Book.class, 1));
+            assertThat(loadedBook)
+                    .isNotNull()
+                    .usingRecursiveComparison()
+                    .withStrictTypeChecking()
+                    .isEqualTo(book);
```

Comment:
If `loadedBook` is `null`, then `isEqualTo` fails. We don't need `isNotNull`.

```suggestion
            assertThat(loadedBook)
                    .usingRecursiveComparison()
                    .withStrictTypeChecking()
                    .isEqualTo(book);
```

Reply:
That is true, but the gist here is more appropriate failure message and code logic readability (`get()` could return null if not found so it is worthwhile to test it as the first step). I might well add message in `isNotNull()` to explain that the book should have been found in db.

Reply:
well, `isNotNull()` doesn't accept message. Could we retain it as the first step if only for testing logic cleaness reason?

Reply:
I think it clutters the assertion without adding any benefit. But if you see it as more readable, let's leave it.